File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let i = 0;
        let j = numbers.length-1;

        while (i < j) {
            const n1 = numbers[i];
            const n2 = numbers[j];
            const sum = n1+n2;

            if (sum > target) {
                j--;
            } else if (sum < target) {
                i++;
            } else {
                return [++i, ++j];
            }
        }
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def isHappy(self, n: int) -> bool:
        s = set()
        while n not in s and n != 1:
            s.add(n)
            n = sum(map(lambda x: x*x, map(int, f"{n}")))

        return True if n == 1 else False